src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -54,18 +54,23 @@ private void refreshCacheItem(final MessageCacheItem cacheItem, Iterator<DataSou
 
 		long timestampOld = cacheItem.getTimestamp();
 		DataSourceEnum dataSource = msgSourceQueueIter.next();
+		String localeOrig = dto.getLocale();
+		if (dataSource.equals(DataSourceEnum.VIP) && dto.getLocale().equals(ConstantsKeys.SOURCE)) {
```

Comment:
Is it better unify the message file suffix in Singleton service and local bundle either to 'latest' or to 'source', then add the unified suffix to the fallback queue, in this case you needn't judge the data source and the requested locale here.

Reply:
Yes, we will do that later on. We shall officially use "source" in the service (new version). Then in the client libraries, we have to support both versions of VIP service (old version with "latest" for backwards compatibility, and new version with "source").

In the meantime, for compatibility with current VIP service, we have to add this line.